storer/files.go

```diff
@@ -50,11 +59,45 @@ func GetTaskFileWriteCloser(id int) (io.WriteCloser, error) {
 			log.Printf("Cannot create directory %s Error %s", dir, err)
 		}
 	}
-	file, err := os.Create(fmt.Sprintf("%s/task-%d", dir, id))
+	file, err := os.Create(getTaskPath(dir, id))
 	if err != nil {
 		log.Printf("Cannot create file task-%d Error %s", id, err)
 		return nil, err
 
 	}
 	return file, nil
 }
+
+func S3UploadTask(bucket string, id int) {
+	dir := viper.GetString(misc.OutDirKey)
+	awsRegion := viper.GetString(misc.AWSRegion)
+	filename := getTaskPath(dir, id)
+	file, err := os.Open(filename)
+	if err != nil {
+		fmt.Println("Failed to open file for upload to S3 bucket", filename, err)
+		os.Exit(1)
+	}
+	defer file.Close()
+
+	//TODO fix credentials
+	conf := aws.Config{
+		Region:      aws.String(awsRegion),
+		Credentials: credentials.NewSharedCredentials("", "production_42"),
+	}
+	sess := session.New(&conf)
+	svc := s3manager.NewUploader(sess)
+	if viper.GetBool(misc.DebugKey) {
+		fmt.Println("Uploading file to S3")
+	}
+	result, err := svc.Upload(&s3manager.UploadInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(filepath.Base(filename)),
+		Body:   file,
+	})
+	if err != nil {
+		fmt.Println("error", err)
+		os.Exit(1)
```

Comment:
It is not good for server application to perform exist if for some reason 1 task was not uploaded to the S3. Because os.Exit(1) will stop tho whole server. For example if some one deactivated the AWS AMI access key, server will stop operating. 
Instead if is a good idea to use log.Printf for generic error (aka INFO level) and use construction if viper.GetBool(DebugKey) { log.Prinf  for more verbose error description in the debug mode.